src/zenml/zen_stores/secrets_stores/sql_secrets_store.py

```diff
@@ -0,0 +1,438 @@
+
+        If the update includes a change of name or scope, the scoping rules
+        enforced in the secrets store are used to validate the update:
+
+          - only one workspace-scoped secret with the given name can exist
+            in the target workspace.
+          - only one user-scoped secret with the given name can exist in the
+            target workspace for the target user.
+
+        Args:
+            secret_id: The ID of the secret to be updated.
+            secret_update: The update to be applied.
+
+        Returns:
+            The updated secret.
+
+        Raises:
+            KeyError: if the secret doesn't exist.
+            EntityExistsError: If a secret with the same name already exists in
+                the same scope.
+        """
+        with Session(self.engine) as session:
+            existing_secret = session.exec(
+                select(SecretSchema).where(SecretSchema.id == secret_id)
+            ).first()
+
+            if not existing_secret:
+                raise KeyError(f"Secret with ID {secret_id} not found.")
+
+            # A change in name or scope requires a check of the scoping rules.
+            if (
+                secret_update.name is not None
+                and existing_secret.name != secret_update.name
+                or secret_update.scope is not None
+                and existing_secret.scope != secret_update.scope
+            ):
+                secret_exists, msg = self._check_secret_scope(
+                    session=session,
+                    secret_name=secret_update.name,
+                    scope=secret_update.scope,
+                    workspace=secret_update.workspace,
+                    user=secret_update.user,
+                    exclude_secret_id=secret_id,
+                )
+                if secret_exists:
+                    raise EntityExistsError(msg)
+
+            existing_secret.update(
+                secret_update=secret_update,
+                encryption_engine=self._encryption_engine,
+            )
+            session.add(existing_secret)
+            session.commit()
+
+            # Refresh the Model that was just created
+            session.refresh(existing_secret)
+            return existing_secret.to_model(
+                encryption_engine=self._encryption_engine
+            )
+
+    @track(AnalyticsEvent.DELETED_SECRET)
+    def delete_secret(self, secret_id: UUID) -> None:
+        """Delete a secret.
+
+        Args:
+            secret_id: The id of the secret to delete.
+
+        Raises:
+            KeyError: if the secret doesn't exist.
+        """
+        with Session(self.engine) as session:
+            try:
+                secret_in_db = session.exec(
+                    select(SecretSchema).where(SecretSchema.id == secret_id)
+                ).one()
+                session.delete(secret_in_db)
+            except NoResultFound as error:
+                raise KeyError from error
```

Comment:
I think the error message isn't really nice when doing this, it'll just be an empty KeyError right